kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/PodPresetTest.java

```diff
@@ -0,0 +1,208 @@
+          .withMountPath("/cache")
+          .withName("cache-volume")
+          .build())
+        .withVolumes(
+          new VolumeBuilder()
+            .withName("cache-volume")
+            .withEmptyDir(new io.fabric8.kubernetes.api.model
+              .EmptyDirVolumeSourceBuilder()
+              .build())
+            .build())
+        .endSpec()
+        .build())
+        .once();
+
+      pod = client.pods().inNamespace("default").withName(pod.getMetadata().getName()).get();
+      assertEquals("pod-test", pod.getMetadata().getName());
+      assertNotNull(pod.getMetadata().getAdditionalProperties().get("DB_PORT"));
+      assertNotNull(pod.getMetadata().getAdditionalProperties().get("cache-volume"));
+
+      PodPreset podpreset = client.settings().podPresets().withName("podpreset-test").get();
+      assertEquals("podpreset-test", podpreset.getMetadata().getName());
+    } catch (Exception e) {
+      e.printStackTrace();
+    } finally {
+      // Delete Pod
+      client.pods().withName("pod-test").delete();
+    }
+  }
+
+  @Test
+  public void testDelete(){
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/test/podpresets/podpreset1").andReturn(200, new PodPresetBuilder()
+      .withNewMetadata().withName("podpreset1").endMetadata()
+      .withNewSpec()
+      .withNewSelector().withMatchLabels(Collections.singletonMap("app", "website")).endSelector()
+      .withEnv(new EnvVarBuilder()
+        .withName("DB_PORT")
+        .withValue("6379")
+        .build())
+      .withVolumeMounts(new VolumeMountBuilder()
+        .withMountPath("/cache")
+        .withName("cache-volume")
+        .build())
+      .withVolumes(
+        new VolumeBuilder()
+          .withName("cache-volume")
+          .withEmptyDir(new io.fabric8.kubernetes.api.model
+            .EmptyDirVolumeSourceBuilder()
+            .build())
+          .build())
+      .endSpec()
+      .build())
+      .once();
+
+    server.expect().withPath("/apis/settings.k8s.io/v1alpha1/namespaces/ns1/podpresets/podpreset2").andReturn(200, new PodPresetBuilder()
+      .withNewMetadata().withName("podpreset2").endMetadata()
+      .withNewSpec()
+      .withNewSelector().withMatchLabels(Collections.singletonMap("app", "website")).endSelector()
+      .withEnv(new EnvVarBuilder()
+        .withName("DB_PORT")
+        .withValue("6379")
+        .build())
+      .withVolumeMounts(new VolumeMountBuilder()
+        .withMountPath("/cache")
+        .withName("cache-volume")
+        .build())
+      .withVolumes(
+        new VolumeBuilder()
+          .withName("cache-volume")
+          .withEmptyDir(new io.fabric8.kubernetes.api.model
+            .EmptyDirVolumeSourceBuilder()
+            .build())
+          .build())
+      .endSpec()
+      .build())
+      .once();
+
+    KubernetesClient client = server.getClient();
+
+    Boolean deleted = client.settings().podPresets().withName("podpreset1").delete();
+    assertNotNull(deleted);
```

Comment:
Why would you do an `assertNotNull` for boolean values ?

Remember.. False is not equal to Null. So regardless of the value being true or false, assertNotNull will always pass.

Reply:
@dev-gaur Oh alright, understood. 
Although assertNull has been used for boolean values in other older tests cases as well. 
In PodTest, ReplicationControllerTest, ReplicaSetTest, PriorityClassTest to name a few. 
Should I fix those changes in this PR or should I open a new PR for them?

Reply:
@dev-gaur Oh alright, understood. 
Although assertNotNull has been used for boolean values in other older tests cases as well. 
In PodTest, ReplicationControllerTest, ReplicaSetTest, PriorityClassTest to name a few. 
I'll refer to those as well in a new PR.

Reply:
thanks for pointing out. we'll appreciate your contributions.

Reply:
I think you forgot fixing this part @kunal-kushwaha

Reply:
> I think you forgot fixing this part @kunal-kushwaha

Oh right, my bad
Fixed